### PR TITLE
Feat(eos_designs): Add support for mlag_peer_ipv6_pool

### PR DESCRIPTION
**Pull request by Xatrekak**
## Change Summary

This PR adds support for configuring the L2 MLAG peer via IPv6. This is accomplished by adding new input-variables and the supporting infrastructure:

- mlag_peer_address_family
- mlag_peer_ipv6_pool
- ipv6_prefix_length

## Related Issue(s)
#3715

## Component(s) name

`arista.avd.eos_designs`
`arista.avd.eos_cli_config_gen` (Documentation only. No logical changes to component.)

## Proposed changes
This changes adds the capability to configure the MLAG peer using IPv6. To accomplish this a new input variable has been created that allows the selection of an address family via `mlag_peer_address_family`.
In addition to this `mlag_peer_ipv6_pool` has been added to specify the IPv6 pool to use.
```yaml
l3leaf:
  defaults:
    mlag_peer_address_family: ipv6
    mlag_peer_ipv6_pool: 2001:DB8::/56
```

 The default subnet when creating peers has been set to `/64` following the current IPv6 best practices, however this is configurable via `ipv6_prefix_length` e.g:
```yaml
fabric_ip_addressing:
  mlag:
    ipv6_prefix_length: 127
```

## How to test
molecules have been created and ran:
`molecule converge -s eos_designs_unit_tests -- --limit MLAG_IPV6`

This has also been ran in a lab environment running vEOS and CVP. This feature has been built and successful tested against a local AVD enviroment setup in this lab.

```EOS
leaf2#show mlag
MLAG Configuration:              
domain-id                          :        DC1_L3_LEAF1
local-interface                    :            Vlan4094
peer-address                       :         2001:db8::1
! Output omitted
MLAG Status:                     
state                              :              Active
negotiation status                 :           Connected
```

## Limitations

This has not been tested in the scenario where both the normal MLAG peer and the L3 MLAG peer both use the same SVI and is likely unsupported currently.
This scenario will be tested and resolved in a future PR that adds support for configuring the L3 MLAG peer with IPv6.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/MLAG_IPV6.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    mlag_peer_ipv6_pool: 2001:DB7::/56
+    mlag_peer_l3_ipv4_pool: 10.10.224.0/24
+    loopback_ipv4_pool: 192.168.255.0/24
+    loopback_ipv4_offset: 32
+    vtep_loopback_ipv4_pool: 192.168.254.0/24
+    platform: vEOS-LAB
+    mlag_interfaces: [Ethernet5, Ethernet6]
+  node_groups:
+    - group: MLAG_IPV6
+      bgp_as: 923
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/MLAG_IPV6.yml]: Please use a private ASN

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml [hunk 2]
```diff
@@ -753,14 +753,32 @@ $defs:
             documentation_options:
               table: node-type-l2-mlag-configuration
             type: str
+          mlag_peer_address_family:
+            documentation_options:
+              table: node-type-l2-mlag-configuration
+            description: |
+              Toggle to override the default (ipv4) address family used to establish MLAG Peer Link (control link).
+              Required when using IPv6 or other non-default values for MLAG leafs present in topology.
+              Requires minimum EOS version of 4.31.1f or higher.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]: Not sure what you mean with `other non-default values`?
```suggestion
            description: |-
              IP address family used to establish MLAG Peer Link (control link).
              `ipv6` requires EOS version 4.31.1F or higher.
              Note: `ipv6` is not supported in combination with a common MLAG peer link VLAN (ex. `mlag_l3_peer_vlan` set to 4094).
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml [hunk 3]
```diff
@@ -753,14 +753,32 @@ $defs:
...
+              - ipv4
+              - ipv6
+            default: ipv4
           mlag_peer_ipv4_pool:
             documentation_options:
               table: node-type-l2-mlag-configuration
             description: |
-              IP address pool used for MLAG Peer Link (control link). IP is derived from the node id.
+              IPv4 address pool used for MLAG Peer Link (control link). IP is derived from the node id.
+              Required when MLAG leafs present in topology.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]:
```suggestion
              Required for MLAG leafs when `mlag_peer_address_family` is `ipv4` (default).
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml [hunk 4]
```diff
@@ -753,14 +753,32 @@ $defs:
...
+              Requires minimum EOS version of 4.31.1f or higher.
+            type: str
+            valid_values:
+              - ipv4
+              - ipv6
+            default: ipv4
           mlag_peer_ipv4_pool:
             documentation_options:
               table: node-type-l2-mlag-configuration
             description: |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]:
```suggestion
            description: |-
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml [hunk 5]
```diff
@@ -753,14 +753,32 @@ $defs:
...
-              IP address pool used for MLAG Peer Link (control link). IP is derived from the node id.
+              IPv4 address pool used for MLAG Peer Link (control link). IP is derived from the node id.
+              Required when MLAG leafs present in topology.
+            type: str
+          mlag_peer_ipv6_pool:
+            documentation_options:
+              table: node-type-l2-mlag-configuration
+            description: |
+              IPv6 address pool used for MLAG Peer Link (control link). IP is derived from the node id.
               Required when MLAG leafs present in topology.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]:
```suggestion
            description: |-
              IPv6 address pool used for MLAG Peer Link (control link). IP is derived from the node id.
              Required for MLAG leafs when `mlag_peer_address_family` is `ipv6`.
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml (hunk 2 above)
Comment (Xatrekak) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]: Manually added in suggestion.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Thanks! Please make sure someone tests this including on EOS so we know the generated config is valid.

**Comment by ClausHolbechArista** [comment]
Since you are changing the order of structured config, you also get the "joy" of having to rerun all our molecule tests locally.
```sh
cd ansible_collections/arista/avd/molecule
make refresh-facts
```

**Review by jerryatarista — APPROVED** [review]
LGTM

**Pull request merged**